pgcli/main.py

```diff
@@ -27,23 +26,44 @@
 from .config import write_default_config, load_config
 from .key_bindings import pgcli_bindings
 
+from urlparse import urlparse
+from getpass import getuser
+from psycopg2 import OperationalError
+
 _logger = logging.getLogger(__name__)
 
 @click.command()
-@click.option('-h', '--host', default='', help='Host address of the '
-        'postgres database.')
+@click.option('-h', '--host', default='localhost', envvar='PGHOST',
+        help='Host address of the postgres database.')
 @click.option('-p', '--port', default=5432, help='Port number at which the '
-        'postgres instance is listening.')
-@click.option('-U', '--user', prompt=True, envvar='USER', help='User name to '
+        'postgres instance is listening.', envvar='PGPORT')
+@click.option('-U', '--user', envvar='PGUSER', help='User name to '
         'connect to the postgres database.')
-@click.option('-W', '--password', is_flag=True, help='Force password prompt.')
-@click.argument('database', envvar='USER')
-def cli(database, user, password, host, port):
-    if password:
+@click.option('-W', '--password', 'prompt_passwd', is_flag=True, default=False,
+              help='Force password prompt.')
+@click.option('-w', '--no-password', 'never_prompt', is_flag=True,
+              default=False, help='Never issue a password prompt')
+@click.argument('database', default='', envvar='PGDATABASE')
+def cli(database, user, host, port, prompt_passwd, never_prompt):
+
+    passwd = ''
+    if not database:
+        #default to current OS username just like psql
+        database = user = getuser()
+    elif '://' in database:
+        #a URI connection string
+        parsed = urlparse(database)
+        database = parsed.path[1:]  # ignore the leading fwd slash
+        user = parsed.username
+        passwd = parsed.password
+        port = parsed.port
+        host = parsed.hostname
+
+    if prompt_passwd and not passwd:
         passwd = click.prompt('Password', hide_input=True, show_default=False,
                 type=str)
-    else:
-        passwd = ''
+
+    auto_passwd_prompt = not passwd and not never_prompt
```

Comment:
I think I understand the logic, but I'd prefer if there was a comment here explaining what this does. Feel free to write a paragraph if you'd like. I don't believe in the idea that a "Well-written code would never need comments". I like to be generous with my commenting. 

A detailed comment today will save me 30mins when I come back to it 2 months later. :)